spring-restdocs-core/src/main/java/org/springframework/restdocs/cli/CommandFormatter.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2014-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.restdocs.cli;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Formatter for {@link CurlRequestSnippet} and {@link HttpieRequestSnippet}.
+ * Its purpose is to create a command snippet with a selected separator.
+ *
+ * @author Tomasz Kopczynski
+ */
+public final class CommandFormatter {
```

Comment:
OK, I think I managed to introduce the `ConcatenatingCommandFormatter` class as you proposed.